packages/forms/frontend/sirius-components-widget-reference/src/components/ValuedReferenceAutocomplete.tsx

```diff
@@ -0,0 +1,173 @@
+  };
+  const classes = useStyles(props);
+
+  const handleAutocompleteChange = (_event, newValue) => {
+    let newValueIds = newValue.map((value) => value.id);
+    if (!widget.reference.manyValued && widget.referenceValues.length > 0) {
+      // For mono-valued reference, we only keep the new one
+      newValueIds = newValueIds.filter((newValue) => widget.referenceValues.some((value) => value.id !== newValue));
+    }
+    const variables = {
+      input: {
+        id: crypto.randomUUID(),
+        editingContextId,
+        representationId: formId,
+        referenceWidgetId: widget.id,
+        newValueIds,
+      },
+    };
+    editReference({ variables });
+  };
+
+  const handleClearClick = () => {
+    const variables = {
+      input: {
+        id: crypto.randomUUID(),
+        editingContextId,
+        representationId: formId,
+        referenceWidgetId: widget.id,
+        newValueIds: [],
+      },
+    };
+    editReference({ variables });
+  };
+
+  let placeholder: string;
+  if (widget.reference.manyValued) {
+    placeholder = 'Values';
+  } else {
+    placeholder = widget.referenceValues.length > 0 ? '' : 'Value';
+  }
+  return (
+    <Autocomplete
+      data-testid={widget.label}
+      multiple
+      disabled={readOnly || widget.readOnly}
+      options={widget.referenceOptions}
+      getOptionLabel={(option: GQLReferenceValue) => option.label}
+      value={widget.referenceValues}
+      getOptionSelected={(option, value) => option.id === value.id}
+      onChange={handleAutocompleteChange}
+      renderOption={(option: GQLReferenceValue) => (
+        <>
+          {option.iconURL ? <img width="16" height="16" alt={''} src={httpOrigin + option.iconURL} /> : null}
+          <span className={classes.optionLabel} data-testid={`option-${option.label}`}>
+            {option.label}
+          </span>
+        </>
+      )}
+      disableClearable
+      renderTags={(value, getTagProps) =>
+        value.map((option, index) => (
+          <Chip
+            key={index}
+            classes={{ label: classes.referenceValueStyle }}
+            label={option.label}
+            data-testid={`reference-value-${option.label}`}
+            icon={option.iconURL ? <img width="16" height="16" alt={''} src={httpOrigin + option.iconURL} /> : null}
+            clickable={!readOnly && !widget.readOnly}
+            onClick={() => optionClickHandler(option)}
+            {...getTagProps({ index })}
+          />
+        ))
+      }
+      renderInput={(params) => (
+        <TextField
+          {...params}
+          variant="standard"
+          placeholder={placeholder}
+          InputProps={{
+            ...params.InputProps,
+            endAdornment: (
+              <>
+                {params.InputProps.endAdornment}
+                <InputAdornment position="end">
+                  <IconButton
+                    aria-label="more"
+                    size="small"
+                    title="Select an object"
```

Comment:
I think "Edit" would be better, as the modal can be used to remove or re-order elements in the case of a multi-valued ref.

Reply:
Done